### PR TITLE
Consolidate Chemistry data wrangling tools into one function

### PR DESCRIPTION
**Pull request by guenp**
One function to load a Broombridge file and generate a tuple to pass as `JWEncodedData`.

### CODE REVIEW

File: Python/qsharp-chemistry/qsharp_chemistry/shims.py [hunk 1]
```diff
@@ -93,3 +93,41 @@ def encode(hamiltonian: FermionHamiltonian, input_state: InputState) -> Tuple:
         input_state=input_state.__dict__
     )
     return data
+
+
+def load_and_encode(
+    file_name: str,
+    problem_description_index: int = 0,
+    initial_state_label: str = None
+) -> Tuple[
```
Comment (msoeken) [inline Python/qsharp-chemistry/qsharp_chemistry/shims.py:102]: Is it possible to define type aliases in Python? This is a little hard to parse.

**Review by msoeken — APPROVED** [review]
Looks good to me, just had one small comment.

File: Python/qsharp-chemistry/qsharp_chemistry/shims.py [hunk 2]
```diff
@@ -93,3 +93,41 @@ def encode(hamiltonian: FermionHamiltonian, input_state: InputState) -> Tuple:
...
+    # TODO: This function accesses a file multiple times, find a way to cache 
+    # or pass data explicitly
+    # TODO: Make these tuples of lists etc. a bit more insightful data types
+    broombridge_data =  load_broombridge(file_name)
+    problem = broombridge_data.problem_description[problem_description_index]
+
+    if initial_state_label is None:
+        # Pick first in list
+        initial_state_label = problem.initial_state_suggestions[0].get("Label")
+        logger.info(f"Using inital state label: {initial_state_label}")
```
Comment (rmshaffer) [inline Python/qsharp-chemistry/qsharp_chemistry/shims.py:127]:
```suggestion
        logger.info(f"Using initial state label: {initial_state_label}")
```

File: Python/qsharp-chemistry/qsharp_chemistry/shims.py (hunk 1 above)
Comment (rmshaffer) [reply Python/qsharp-chemistry/qsharp_chemistry/shims.py:102]: I just recently discovered: https://docs.python.org/3/library/typing.html#type-aliases

File: Python/qsharp-chemistry/qsharp_chemistry/problem_description.py [hunk 3]
```diff
@@ -35,10 +36,24 @@ def __eq__(self, other):
...
+    geometry: dict = field(default_factory=dict)
+    coulomb_repulsion: dict = field(default_factory=dict)
+    scf_energy: dict = field(default_factory=dict)
+    scf_energy_offset: dict = field(default_factory=dict)
+    fci_energy: dict = field(default_factory=dict)
+    n_orbitals: int = 0
+    n_electrons: int = 0
+    energy_offset: dict = field(default_factory=dict)
+    hamiltonian: dict = field(default_factory=dict)
+
```
Comment (rmshaffer) [inline Python/qsharp-chemistry/qsharp_chemistry/problem_description.py:56]: This is a nice idea! Are these used anywhere?

**Review by rmshaffer — APPROVED** [review]

File: Python/qsharp-chemistry/qsharp_chemistry/problem_description.py (hunk 3 above)
Comment (guenp) [reply Python/qsharp-chemistry/qsharp_chemistry/problem_description.py:56]: Good point, right now I just override the `__dict__` property in the constructor, which is kind of a hack... I will make sure to explicitly set each value by grabbing them from the dict.

File: Python/qsharp-chemistry/qsharp_chemistry/shims.py (hunk 1 above)
Comment (guenp) [reply Python/qsharp-chemistry/qsharp_chemistry/shims.py:102]: Will do, thanks both!
